src/DataAccess/ModerationReasonRepository.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\MembershipContext\DataAccess;
+
+use Doctrine\ORM\EntityManager;
+use WMDE\Fundraising\MembershipContext\Domain\Model\ModerationReason;
+
+class ModerationReasonRepository {
+
+	public function __construct(
+		private EntityManager $entityManager
+	) {
+	}
+
+	public function getModerationReasonsThatAreAlreadyPersisted( ModerationReason ...$moderationReasons ): array {
+		if ( count( $moderationReasons ) === 0 ) {
+			return [];
+		}
+		$queryBuilder = $this->entityManager->createQueryBuilder();
+		$queryBuilder->select( 'mr' )
+			->from( ModerationReason::class, 'mr' );
+
+		$paramCounter = 1;
+
+		foreach ( $moderationReasons as $reason ) {
+
+			$condition1 = $queryBuilder->expr()->eq( 'mr.moderationIdentifier', "?$paramCounter" );
+			$queryBuilder->setParameter( $paramCounter, $reason->getModerationIdentifier()->name );
+
+			$paramCounter++;
+			$condition2 = $queryBuilder->expr()->eq( 'mr.source', "?$paramCounter" );
```

Comment:
I think the empty line needs to move down 1 line. The pattern here is "create condition, add parameter, increase paramcounter".

```suggestion
			$queryBuilder->setParameter( $paramCounter, $reason->getModerationIdentifier()->name );
			$paramCounter++;

			$condition2 = $queryBuilder->expr()->eq( 'mr.source', "?$paramCounter" );
```